.github/workflows/ci.yml

```
@@ -0,0 +1,46 @@
+name: Go Project CI
+
+concurrency:
+  group: ci-workflow-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```

```suggestion
      - main
      - release-v**
```

To release a new version, we will first create a branch (e.g: release-v1.0), and will release from that branch.
So I think it's better to also run tests for PR on those branches